src/runtime/plan/suggestion.ts

```diff
@@ -102,12 +102,39 @@ export class Suggestion {
     return (this.hash === other.hash) && (this.descriptionText === other.descriptionText);
   }
 
+  isEqual(other: Suggestion): boolean {
+    return this.isEquivalent(other) &&
+          this.rank === other.rank &&
+          this._isSameSearch(other) &&
+          this._isSameDescription(other) &&
+          this._isSameVersions(other);
+  }
+
+  _isSameSearch(other: Suggestion): boolean {
+    return this.searchGroups.length === other.searchGroups.length &&
+           this.searchGroups.every(search => other.hasSearchGroup(search));
+  }
+
+  _isSameDescription(other: Suggestion): boolean {
+    return Object.keys(this.descriptionByModality).length === Object.keys(other.descriptionByModality).length &&
+        Object.keys(this.descriptionByModality).every(
+            key => JSON.stringify(this.descriptionByModality[key]) === JSON.stringify(other.descriptionByModality[key]));
+  }
+
+  _isSameVersions(other: Suggestion): boolean {
+    return Object.keys(this.versionByStore).length === Object.keys(other.versionByStore).length &&
```

Comment:
versionByStore might benefit from being a Map too?

Reply:
comparing 2 Maps might be a tiny bit shorter, but would still need a helper method, i think. 
is there a benefit in using Map over {}?